instabot/bot/bot.py

```diff
@@ -236,41 +230,43 @@ def login(self, **args):
     def prepare(self):
         storage = load_checkpoint(self)
         if storage is not None:
-            (self.total_liked,
-             self.total_unliked,
-             self.total_followed,
-             self.total_unfollowed,
-             self.total_commented,
-             self.total_blocked,
-             self.total_unblocked,
-             self.api.total_requests,
-             self.start_time,
-             self.total_archived,
-             self.total_unarchived,
-             self.total_sent_messages) = storage
+            self.total, self.api.total_requests, self.start_time = storage
 
     def print_counters(self):
-        if self.total_liked:
-            self.logger.info("Total liked: %d", self.total_liked)
-        if self.total_unliked:
-            self.logger.info("Total unliked: %d", self.total_unliked)
-        if self.total_followed:
-            self.logger.info("Total followed: %d", self.total_followed)
-        if self.total_unfollowed:
-            self.logger.info("Total unfollowed: %d", self.total_unfollowed)
-        if self.total_commented:
-            self.logger.info("Total commented: %d", self.total_commented)
-        if self.total_blocked:
-            self.logger.info("Total blocked: %d", self.total_blocked)
-        if self.total_unblocked:
-            self.logger.info("Total unblocked: %d", self.total_unblocked)
-        if self.total_archived:
-            self.logger.info("Total archived: %d", self.total_archived)
-        if self.total_unarchived:
-            self.logger.info("Total unarchived: %d", self.total_unarchived)
-        if self.total_sent_messages:
-            self.logger.info("Total sent messages: %d", self.total_sent_messages)
-        self.logger.info("Total requests: %d", self.api.total_requests)
+        for key, val in self.total.items():
+            if val > 0:
+                self.logger.info("Total {}: {}".format(key, val))
+        self.logger.info("Total requests: {}".format(self.api.total_requests))
+
+    def delay(self, key):
+        """Sleep only if elapsed time since `self.last[key]` < `self.delay[key]`."""
+        last_action, target_delay = self.last[key], self.delays[key]
+        elapsed_time = time.time() - last_action
+        if elapsed_time < target_delay:
+            t_remaining = target_delay - elapsed_time
+            time.sleep(t_remaining * random.uniform(0.25, 1.25))
+        self.last[key] = time.time()
+
+    def error_delay(self):
+        time.sleep(10)
+
+    def small_delay(self):
+        time.sleep(random.uniform(0.75, 3.75))
+
+    def very_small_delay(self):
+        time.sleep(random.uniform(0.175, 0.875))
+
+    def reached_limit(self, key):
+        current_date = datetime.datetime.now()
+        passed_days = (current_date.date() - self.start_time.date()).days
+        if passed_days > 1:
```

Comment:
I think, there should be:
```python 
if passed_days > 0:
```
otherwise, a user must wait yet another day to reach a limit